core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
+    return CurrentTagsUtils.getCurrentTagContext();
   }
 
   /**
-   * Returns an empty {@code TagContext}.
+   * Returns a new empty {@code Builder}.
    *
-   * @return an empty {@code TagContext}.
+   * @return a new empty {@code Builder}.
    */
-  public static TagContext empty() {
-    return EMPTY;
+  public static Builder emptyBuilder() {
+    return new Builder();
   }
 
   /**
-   * Returns an empty {@code Builder}.
+   * Returns a new {@code Builder} created from the current {@code TagContext}.
    *
-   * @return an empty {@code Builder}.
+   * @return a new {@code Builder} created from the current {@code TagContext}.
    */
-  public static Builder newBuilder() {
-    return new Builder();
+  public static Builder currentBuilder() {
```

Comment:
The builder logic shouldn't change at all. Overall, this class shouldn't need to be changed at all.